### PR TITLE
Views: Move 'operation' check to ViewVersion

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: api/src/main/java/org/apache/iceberg/view/ViewVersion.java [hunk 1]
```diff
@@ -65,8 +66,14 @@ public interface ViewVersion {
    *
    * @return the string operation which produced the view version
    */
-  @Value.Derived
+  @Value.Lazy
```
Comment (nastra) [inline api/src/main/java/org/apache/iceberg/view/ViewVersion.java:69]: `@Value.Derived` is eagerly computing the value before `check()` is executed, so we're switching to `@Value.Lazy`

**Review by jackye1995 — APPROVED** [review]

**Comment by jackye1995** [comment]
Looks like there are some merge conflicts, could you rebase?

**Comment by jackye1995** [comment]
Thanks for the fix!

File: api/src/main/java/org/apache/iceberg/view/ViewVersion.java (hunk 1 above)
Comment (rdblue) [reply api/src/main/java/org/apache/iceberg/view/ViewVersion.java:69]: @nastra, why does this matter? All that would happen is that this is set to `null` and then the check method throws an exception.

**Pull request merged**